src/screens/AffiliationViews/DatabaseView/RestorableDatabaseSchemaUpdateDialog.tsx

```diff
@@ -0,0 +1,261 @@
+import * as React from 'react';
+import palette from '@skatteetaten/frontend-components/utils/palette';
+import styled from 'styled-components';
+import {
+  IRestorableDatabaseSchemaData,
+  IUpdateDatabaseSchemaInputWithCreatedBy,
+  IDatabaseSchema,
+  ITestJDBCResponse,
+  IChangeCooldownDatabaseSchemasResponse,
+} from 'models/schemas';
+import { useState, useEffect, useRef } from 'react';
+import Dialog from '@skatteetaten/frontend-components/Dialog';
+import { Grid } from '@skatteetaten/frontend-components/Grid';
+import { getLocalDatetime } from 'utils/date';
+import JdbcConnection from './JdbcConnection';
+import { TextFieldEvent } from 'types/react';
+import Labels from './Labels';
+import ConfirmationDialog from 'components/ConfirmationDialog';
+import Button from '@skatteetaten/frontend-components/Button';
+import ActionButton from '@skatteetaten/frontend-components/ActionButton';
+
+const { skeColor } = palette;
+
+interface IRestorableDatabaseSchemaUpdateDialogProps {
+  schema?: IRestorableDatabaseSchemaData;
+  className?: string;
+  testJdbcConnectionResponse: ITestJDBCResponse;
+  restoreResponse: IChangeCooldownDatabaseSchemasResponse;
+  clearSelectedSchema: () => void;
+  onUpdate: (databaseSchema: IUpdateDatabaseSchemaInputWithCreatedBy) => void;
+  onDelete: (databaseSchema: IDatabaseSchema) => void;
+  onTestJdbcConnectionForId: (id: string) => void;
+  onRestoreDatabaseSchema: (schema: IDatabaseSchema, active: boolean) => void;
+}
+
+interface IUpdatedSchemaValues {
+  id: string;
+  discriminator: string;
+  createdBy: string;
+  engine: string;
+  description: string;
+  environment: string;
+  application: string;
+  affiliation: string;
+}
+
+function RestorableDatabaseSchemaUpdateDialog({
+  schema,
+  clearSelectedSchema,
+  className,
+  onTestJdbcConnectionForId,
+  testJdbcConnectionResponse,
+  onRestoreDatabaseSchema,
+  restoreResponse,
+}: IRestorableDatabaseSchemaUpdateDialogProps) {
+  const initialUpdatedSchemaValues: IUpdatedSchemaValues = {
+    id: '',
+    discriminator: '',
+    createdBy: '',
+    engine: '',
+    description: '',
+    environment: '',
+    application: '',
+    affiliation: '',
+  };
+  const [updatedSchemaValues, setUpdatedSchemaValues] = useState<
+    IUpdatedSchemaValues
+  >(initialUpdatedSchemaValues);
+
+  function usePrevious(value) {
+    const ref = useRef();
+    useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  }
+
+  const handleLabelChange = (field: string) => (
```

Comment:
Er denne funksjonen korrekt? Ser at typescript kaster kompileringsfeil på grunn av feil type for setUpdatedSchemaValues()
Den samme funksjonen i databaseSchemaupdatedialog kaster ikke typefeil.